test/linarith.lean

```diff
@@ -347,3 +348,19 @@ by linarith
 example (p q r s t u v w : ℕ) (h1 : p + u = q + t) (h2 : r + w = s + v) :
   p * r + q * s + (t * w + u * v) = p * s + q * r + (t * v + u * w) :=
 by nlinarith
+
+-- Tests involving a norm, including that squares in a type where `pow_two_nonneg` does not apply
+-- do not cause an exception
+variables {R : Type*} [normed_ring R]
+
+example (t : R) (a b : ℝ) (h : a ≤ b) : ∥t^2∥ * a ≤ ∥t^2∥ * b :=
+mul_le_mul_of_nonneg_left h (norm_nonneg (t^2))
```

Comment:
I don't think tests 1, 3 and 4 are necessary, I wrote them on Zulip just to pin down the source of the error to `nlinarith`.  Test 1 doesn't invoke `linarith`!

Reply:
Oh I agree, but I just figured the more tests for (n)linarith the better! But I didn't spot the first one though so I'll remove that one only unless someone objects to having more tests like this.